packages/lib/src/link/connectors.ts

```diff
@@ -9,6 +9,25 @@ import { logger } from "../logging";
 import ExponentialBackoff from "../ExponentialBackoff";
 import type { Request, Event } from "./link";
 
+/**
+ * Numbered codes describing why a connection was closed
+ * See: https://github.com/Luka967/websocket-close-codes
+ */
+export enum ConnectionClosed {
+	// Codes between 1000 and 1015 are pre-allocated for web sockets
+	Normal = 1000, // CLOSE_NORMAL
+	ServerQuit = 1001, // CLOSE_GOING_AWAY
+	ProtocolError = 1002, // CLOSE_PROTOCOL_ERROR
+	PolicyError = 1008, // Policy Violation
+	ServerError = 1011, // Server Error
+
+	// Codes after 4000 are available for applications
+	// Ours are somewhat based on http client error codes
+	BadRequest = 4000,
```

Comment:
While we're inspired by HTTP, I don't think copying out-right makes sense. The "Request" here would be referring to receiving a malformed WebSocket message, which is not the same as a request in http, and not the same as what "Request" refers to in link layer.

Perhaps call it BadMessage? Or MalformedMessage.

Reply:
I like the sound of Malformed Message, I'll update to that.